securing_apps/topics/oidc/javascript-adapter.adoc

```diff
@@ -161,6 +161,59 @@ For the Hybrid flow, you need to pass the parameter `flow` with value `hybrid` t
 keycloak.init({ flow: 'hybrid' })
 ----
 
+==== Hybrid Apps with Cordova
+
+Keycloaks support hybird mobile apps developed with https://cordova.apache.org/[Apache Cordova]. The Javascript adapter has two modes for this: `cordova` and `cordova-native`:
+
+The default is `cordova`. The adapter will pick it automatically if no adapter type has been configured and `window.cordova` is present. 
+When logging in, it will open an https://cordova.apache.org/docs/en/latest/reference/cordova-plugin-inappbrowser/[InApp Browser] that lets the user interact with keycloak and afterwards returns to the app by redirecting to http://localhost. 
+Therefore this URL needs to be whitelisted as valid redirect-uri in the client configuration section of the Administration Console.
+
+While this mode is very easy to setup, it also has a few disadvantages:
+The InApp-Browser is a browser embedded in the app and not the phones default browser. 
+Therefore it will have different settings and stored credentials will not be available. It might be slower, especially when rendering more complex themes.
+There are security concerns to consider, before using this mode: 
+It is possible for the app to gain access to the credentials of the user, as it has full control of the browser rendering the login page.
+So do not allow its use in apps you dont trust.
+
+We have an example app that can help you get started: https://github.com/keycloak/keycloak/tree/master/examples/cordova
+
+The alternative mode `cordova-native` takes a different approach.
+It opens the login page in the systems browser. 
+After the user has authenticated, the browser redirects back into the app by a special URL.
+From there, the keycloak adapter can finish the login by reading the code or token from the URL.
```

Comment:
The alternative mode `cordova-native` takes a different approach. It opens the login page using the system's browser. After the user has authenticated, the browser redirects back into the app using a special URL. From there, the {project_name} adapter can finish the login by reading the code or token from the URL.